windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1115 @@
+}
+
+
+//
+// RNFS implementations
+//
+void RNFSManager::ConstantsViaConstantsProvider(RN::ReactConstantProvider& constants) noexcept
+{
+    // RNFS.MainBundlePath
+    constants.Add(L"RNFSMainBundlePath", to_string(Package::Current().InstalledLocation().Path()));
+
+    // RNFS.CachesDirectoryPath
+    constants.Add(L"RNFSCachesDirectoryPath", to_string(ApplicationData::Current().LocalCacheFolder().Path()));
+
+    // RNFS.ExternalCachesDirectoryPath - NULL in iOS (No equivalent in Windows)
+
+    // RNFS.DocumentDirectoryPath
+    constants.Add(L"RNFSDocumentDirectoryPath", to_string(ApplicationData::Current().LocalFolder().Path()));
+
+    // RNFS.DownloadDirectoryPath - IMPLEMENT for convenience? (absent in iOS and deprecated in Android)
+    constants.Add(L"RNFSDownloadDirectoryPath", UserDataPaths::GetDefault().Downloads());
+
+    // RNFS.ExternalDirectoryPath - NULL in iOS (Pending use case in Windows and deprecated in Android)
+    constants.Add(L"RNFSExternalDirectoryPath", UserDataPaths::GetDefault().Documents());
+
+    // RNFS.ExternalStorageDirectoryPath - NULL in iOS (Pending use case in Windows and deprecated in Android)
+
+    // RNFS.TemporaryDirectoryPath
+    constants.Add(L"RNFSTemporaryDirectoryPath", to_string(ApplicationData::Current().TemporaryFolder().Path()));
+
+    // RNFS.LibraryDirectoryPath - NULL in Android (No equivalent in Windows)
+
+    // RNFS.PicturesDirectoryPath - IMPLEMENT for convenience? (absent in iOS though and deprecated in Android)
+    constants.Add(L"RNFSPicturesDirectoryPath", UserDataPaths::GetDefault().Pictures());
+
+    // RNFS.FileProtectionKeys - NULL in Android (No equivalent in Windows)
+
+    // TODO: Check to see if these can be accessed after package created
+    // Needed for synchronization across Windows devices
+    constants.Add(L"RNFSRoamingDirectoryPath", to_string(ApplicationData::Current().RoamingFolder().Path()));
+
+    // Filetypes
+    constants.Add(L"RNFSFileTypeRegular", 0);
+    constants.Add(L"RNFSFileTypeDirectory", 1);
+}
+
+winrt::fire_and_forget RNFSManager::mkdir(std::string directory, RN::JSValueObject options, RN::ReactPromise<void> promise) noexcept
+try
+{
+    size_t pathLength{ directory.length() };
+
+    if (pathLength <= 0) {
+        promise.Reject("Invalid path.");
+    }
+    else {
+        bool hasTrailingSlash{ directory[pathLength - 1] == '\\' || directory[pathLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? directory.substr(0, pathLength - 1) : directory);
+        path.make_preferred();
+
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(path.parent_path().wstring()) };
+        co_await folder.CreateFolderAsync(path.filename().wstring(), CreationCollisionOption::FailIfExists);
+
+        promise.Resolve();
+    }
+}
+catch (const hresult_error& ex)
+{
+    // "Unexpected error while making directory."
+    promise.Reject( winrt::to_string(ex.message()).c_str() );
```

Comment:
nit can probably avoid the `c_str()`

Reply:
Unfortunately, promise.Reject() does not accept a std::string. It fails to build when I do not use the c_str(). Would it be better to return ReactErrors instead?